manifests/unit.pp

```diff
@@ -22,55 +22,69 @@
   }
-  if $unit_description != undef {
-    validate_string($unit_description)
+  if $unit_description != undef and is_string($unit_description) == false {
+    fail("systemd::${name}::unit::unit_description is not a string")
   }
-  if $environment != undef {
-    validate_string($environment)
+  if $environment != undef and is_string($environment) == false {
+    fail("systemd::${name}::unit::environment is not a string")
   }
-  if $group != undef {
-    validate_string($group)
+  if $group != undef and is_string($group) == false {
+    fail("systemd::${name}::unit::group is not a string")
   }
-  if $user != undef {
-    validate_string($user)
+  if $user != undef and is_string($user) == false {
+    fail("systemd::${name}::unit::user is not a string")
   }
-  if $workingdirectory != undef {
-    validate_string($workingdirectory)
+  if $service_restart != undef and is_string($service_restart) == false {
+    fail("systemd::${name}::unit::restart is not a string")
   }
-  validate_re($ensure, [ '^simple$', '^forking$', '^oneshot$', '^dbus$', '^notify$', '^idle$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
-  if $service_timeoutstartsec != undef {
-    validate_string($service_timeoutstartsec)
+  if $service_execstart != undef and is_string($service_execstart) == false {
+    fail("systemd::${name}::unit::service_execstart is not a string")
   }
-  if $service_restart != undef {
-    validate_string($service_restart)
+  if $service_execstop != undef and is_string($service_execstop) == false {
+    fail("systemd::${name}::unit::service_execstop is not a string")
   }
-  if $service_restartsec != undef {
-    validate_string($service_restartsec)
+  if $install_wantedby != undef and is_string($install_wantedby) == false {
+    fail("systemd::${name}::unit::install_wantedby is not a string")
   }
-  if $service_execstartpre != undef {
-    validate_array($service_execstartpre)
+  if $workingdirectory != undef and is_string($workingdirectory) == false {
+    fail("systemd::${name}::unit::workingdirectory is not a string")
   }
-  if $service_execstart != undef {
-    validate_string($service_execstart)
+
+  if $service_execstartpre != undef and is_array($service_execstartpre) == false {
+    fail("systemd::${name}::unit::service_execstartpre is not an Array")
   }
-  if $service_execstop != undef {
-    validate_string($service_execstop)
+
+  if is_string($service_type) {
+    validate_re($service_type, [ '^simple$', '^forking$', '^oneshot$', '^dbus$', '^notify$', '^idle$' ],
+      "systemd::unit::${name}::ensure is not a string or is invalid and does not match the regex.")
+  } else {
+    fail("systemd::unit::${name}::ensure is not a string or is invalid and does not match the regex.")
   }
-  if $install_wantedby != undef {
-    validate_string($install_wantedby)
+
+  if $service_timeoutstartsec != undef {
+    $service_timeoutstartsec_int = floor($service_timeoutstartsec)
```

Comment:
the string type is the one wanted i think. According to the [doc](https://www.freedesktop.org/software/systemd/man/systemd.service.html) values could be `5min 20s` or `infinity`, ...

Reply:
Hm. Thanks for letting me know! I'll look into it. It was done this way because a module which uses this module as a dependency (written by you as well) sets this value to an integer and not string.
I'll look into how to get it to work in both cases.

Reply:
which module?

Reply:
The PUA module.